packages/autorest.python/autorest/codegen/models/utils.py

```diff
@@ -21,3 +21,10 @@ def add_to_pylint_disable(curr_str: str, entry: str) -> str:
     if curr_str:
         return f"{curr_str},{entry}"
     return f"  # pylint: disable={entry}"
+
+
+def typing_name(
+    file_name: str, internal: bool, need_module_name: bool, type_name: str
+) -> str:
+    module = "_models." if need_module_name else ""
+    return module + (f"{file_name}." if internal else "") + type_name
```

Comment:
a little confused for this function, why `_models.` is hard coded but `file_name` is passed?

Reply:
`why _models. is hard coded but file_name is passed?`
-> For type annotation, the `models` will always imported as `_models` so it is hard coded. `file_name` for msrest mode is `_xxx_models.py` while for dpg mode is `_models.py`. typing_name is util function for both msrest and dpg so file_name is necessary.

Reply:
`why need_module_name add _models and internal add file_name?`
-> In short, need_module_name is used here which make the util function more general.

![image](https://github.com/Azure/autorest.python/assets/70930885/0059bc33-fac6-4733-b8ac-fea6eb371e03)